### PR TITLE
Implement the container image card

### PR DESCRIPTION
**Pull request by cescoffier**

### CODE REVIEW

File: extensions/container-image/runtime/pom.xml [hunk 1]
```diff
@@ -25,6 +25,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-core</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-mutiny</artifactId>
+        </dependency>
```
Comment (gsmet) [inline extensions/container-image/runtime/pom.xml:31]: Do we really want to add a runtime dependency just for the sake of the Dev UI?

File: bom/application/pom.xml [hunk 2]
```diff
@@ -3397,6 +3397,12 @@
                 <version>${vaadin.version}</version>
                 <scope>runtime</scope>
             </dependency>
+            <dependency>
+                <groupId>org.mvnpm.at.vaadin</groupId>
+                <artifactId>select</artifactId>
+                <version>${vaadin.version}</version>
+                <scope>runtime</scope>
+            </dependency>
```
Comment (phillip-kruger) [inline bom/application/pom.xml:3405]: This is already in the bom (scroll down)

File: extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/devconsole/ContainerImageDevUiProcessor.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    CardPageBuildItem create(List<AvailableContainerImageExtensionBuildItem> extensions) {
+        // Get the list of builders
+        JsonArray array = extensions.stream().map(AvailableContainerImageExtensionBuildItem::getName).sorted()
+                .collect(JsonArray::new, JsonArray::add, JsonArray::addAll);
+
+        CardPageBuildItem card = new CardPageBuildItem("Container Image");
+        card.addBuildTimeData("builderTypes", array);
+        card.addPage(Page.webComponentPageBuilder()
+                .title("Build Container")
+                .componentLink("build-container-component.js")
```
Comment (phillip-kruger) [inline extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/devconsole/ContainerImageDevUiProcessor.java:35]: component should follow the conversion qwc-extensionName-pageName so `qwc-container-image-build`

File: extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/devconsole/ContainerImageDevUiProcessor.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                .icon("font-awesome-solid:box"));
+        return card;
+    }
+
+    @BuildStep(onlyIf = IsDevelopment.class)
+    JsonRPCProvidersBuildItem createJsonRPCServiceForContainerBuild() {
+        DevConsoleManager.register("container-build-action", build());
+        return new JsonRPCProvidersBuildItem("ContainerBuilder", ContainerBuilderJsonRpcService.class);
```
Comment (phillip-kruger) [inline extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/devconsole/ContainerImageDevUiProcessor.java:43]: For consistency use the Extension name, so `Container Image`

File: extensions/container-image/deployment/src/main/resources/dev-ui/container-image/build-container-component.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+import '@vaadin/progress-bar';
+import '@vaadin/checkbox';
+import '@vaadin/select';
+import '@vaadin/item';
+import '@vaadin/list-box';
+
+
+export class BuildContainerComponent extends LitElement {
+
+    jsonRpc = new JsonRpc("ContainerBuilder");
```
Comment (phillip-kruger) [inline extensions/container-image/deployment/src/main/resources/dev-ui/container-image/build-container-component.js:21]: Use `Container Image`

File: extensions/container-image/deployment/src/main/resources/dev-ui/container-image/build-container-component.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+import '@vaadin/text-area';
+import '@vaadin/form-layout';
+import '@vaadin/progress-bar';
+import '@vaadin/checkbox';
+import '@vaadin/select';
+import '@vaadin/item';
+import '@vaadin/list-box';
+
+
+export class BuildContainerComponent extends LitElement {
```
Comment (phillip-kruger) [inline extensions/container-image/deployment/src/main/resources/dev-ui/container-image/build-container-component.js:19]: Use the conversion, so `QwcContainerImageBuild`

**Review by phillip-kruger — CHANGES_REQUESTED** [review]

File: extensions/container-image/runtime/pom.xml (hunk 1 above)
Comment (cescoffier) [reply extensions/container-image/runtime/pom.xml:31]: Well, I need to stream, and there is not much I can do without Mutiny. I don't believe it's a problem, as mutiny will likely be there.

File: extensions/container-image/deployment/src/main/resources/dev-ui/container-image/build-container-component.js (hunk 5 above)
Comment (cescoffier) [reply extensions/container-image/deployment/src/main/resources/dev-ui/container-image/build-container-component.js:21]: I would avoid spaces, I went for ContainerImage

Comment (phillip-kruger) [reply extensions/container-image/deployment/src/main/resources/dev-ui/container-image/build-container-component.js]: What I did with using this in a path is to do toLowerCase and replace space with -. Maybe this is better ? Again, I want to be able to do this automatically in the future from the UI, so need some pattern to know how to get the jsonrpc. Also, spaces is ok :)

**Review by phillip-kruger — APPROVED** [review]

**Pull request merged**